packages/qdrant-loader/src/qdrant_loader/config/state.py

```
@@ -112,6 +122,15 @@ def validate_database_path(cls, v: str, info: ValidationInfo) -> str:
         # Return the original value to preserve any environment variables
         return v
 
+    @field_validator("database_url")
+    @classmethod
+    def validate_database_url(cls, v: str | None) -> str | None:
+        if v is not None and "://" not in v:
+            raise ValueError(
+                "database_url must be a full SQLAlchemy URL (scheme://...)"
+            )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)